homeassistant/components/assist_pipeline/__init__.py

```diff
@@ -80,3 +80,35 @@ async def async_pipeline_from_audio_stream(
     )
     await pipeline_input.validate()
     await pipeline_input.execute()
+
+
+async def async_pipeline_detect_wake_word(
```

Comment:
Let's merge this into `async_pipeline_from_audio_stream` by adding option to specify a `start_stage` and set it to `PipelineStage.WAKE_WORD`. We can then re-use all the different features:

 - we get a websocket API
 - we have event callbacks